### ISSUE TITLE
WISDOM_ONLY fails for large input

### ISSUE THREAD

**Issue by OverLordGoldDragon** [pyFFTW/pyFFTW#306]
```python
import pyfftw
import numpy as np

def get_fft_object(x, axis=-1, flag='FFTW_PATIENT'):
    a = pyfftw.empty_aligned(x.shape, dtype=x.dtype)
    b = pyfftw.empty_aligned(x.shape, dtype=x.dtype)
    return pyfftw.FFTW(a, b, axes=(axis,), flags=[flag], planning_timelimit=120, threads=8)

x = np.random.randn(400, 200000).astype('float32')
get_fft_object(x, -1)()
get_fft_object(x, -1, 'FFTW_WISDOM_ONLY')()
```
```python
  File "pyfftw\pyfftw.pyx", line 1280, in pyfftw.pyfftw.FFTW.__cinit__
RuntimeError: No FFTW wisdom is known for this plan.
```

Yet strangely if called with `'PATIENT'` again, it finishes in mere seconds, so seems to be using wisdom under the hood, yet errors when told to do so. Otherwise works fine with e.g. `(400, 2000)` input shape.

**Comment by hgomersall** [comment]
I think what is happening is you are timing out at 120s, then you're calling it _without_ `FFTW_PATIENT` (that flag gets replaced by `FFTW_WISDOM_ONLY`). The planner then completes quickly because it is able to use all the accumulated wisdom and it will now be using `FFT_MEASURE`, which is quicker.

**Comment by OverLordGoldDragon** [comment]
@hgomersall Note `flag='FFTW_PATIENT'` default in `get_fft_object()`, so _without_ is _with_. The error occurs on the second call, `get_fft_object(x, -1, 'FFTW_WISDOM_ONLY')()`, so it fails with `WISDOM_ONLY` after `PATIENT` times out. And despite timing out, a call to `get_fft_object(x, -1)` (which is with `PATIENT`) is much quicker.

**Comment by hgomersall** [comment]
> @hgomersall Note `flag='FFTW_PATIENT'` default in `get_fft_object()`, so _without_ is _with_.

That might be the intent, but that's not what your code above is doing.

**Comment by OverLordGoldDragon** [comment]
@hgomersall Unsure what you mean, but edited code for clarity.

**Comment by hgomersall** [comment]
> @hgomersall Unsure what you mean, but edited code for clarity.

Ok, so now it should be clear: `FFTW_PATIENT` is not being used when you're using `FFTW_WISDOM_ONLY`. The default flag is `FFTW_MEASURE`.

**Comment by hgomersall** [comment]
> I don't follow, so `WISDOM_ONLY` works only _alongside_ other flags? Even in this case, it fails; updated code.

No, `FFTW_WISDOM_ONLY` works in all cases, but it only works if the wisdom for the chosen planning strategy is available to it. If you don't choose a planning strategy, then the default is used, which is `FFTW_MEASURE`. If you limit the time it can plan for and it times-out, then it doesn't necessarily complete so the necessary wisdom is not available to it next time you try that same strategy, which is what is happening with you.

How it handles the case where you timed out and then try calling again is down to `FFTW`. I don't know what policy it will use.

I suggest combining `FFTW_WISDOM_ONLY` with a timeout on the first run is the problem here and shouldn't be done.

**Comment by OverLordGoldDragon** [comment]
So if it times out it doesn't store wisdom? I tested this, `pyfftw.forget_wisdom()`, and `pyfftw.export_wisdom()` was empty, then the command took much longer to run, and when it finished, `pyfftw.export_wisdom()` was no longer empty. Then when I run again with `PATIENT`, it's again in seconds. So... is there something other than wisdom that accelerates compute? For clarity, again, the steps are:

 1. Restart Python kernel
 2. `FFTW_PATIENT` 120 secs, timeout
 3. `FFTW_WISDOM_ONLY` error
 4. `FFTW_PATIENT` 2-3 secs

**Comment by hgomersall** [comment]
This is really an `FFTW` issue. I suggest asking there to understand better. I do think though that timing out is likely to cause complications. It might (though I'm not aware of it) have some stored notion of what was computed in a given time, so it can just pass it back. What happens if you change the time out the second time?

**Comment by OverLordGoldDragon** [comment]
Strangely, nothing happens with `planning_timelimit` set to `119` (1 less) or `121` (1 more), but full recomputation triggers with `240`. Further, I went ahead and `pyfftw.import_wisdom()` from exported saved file, and it worked on the timed out `PATIENT`. Guess this solves my problem, unless this implicit default behavior will change in the future. Opened [fftw3 Issue](https://github.com/FFTW/fftw3/issues/231).
